startup_scripts/220_ip_addresses.py

```diff
@@ -0,0 +1,61 @@
+from ipam.models import IPAddress, VRF
+from dcim.models import Device, Interface
+from virtualization.models import VirtualMachine
+from tenancy.models import Tenant
+from extras.models import CustomField, CustomFieldValue
+from ruamel.yaml import YAML
+
+from netaddr import IPNetwork
+from pathlib import Path
+import sys
+
+file = Path('/opt/netbox/initializers/ip_addresses.yml')
+if not file.is_file():
+  sys.exit()
+
+with file.open('r') as stream:
+  yaml = YAML(typ='safe')
+  ip_addresses = yaml.load(stream)
+
+  optional_assocs = {
+    'tenant': (Tenant, 'name'),
+    'vrf': (VRF, 'name'),
+    'interface': (Interface, 'name')
+  }
+
+  if ip_addresses is not None:
+    for params in ip_addresses:
+      vm = params.pop('virtual_machine', None)
+      device = params.pop('device', None)
```

Comment:
What if someone accidentally puts both, `vm` and `device`? I believe that case should be an error and indicated as such, i.e. the script should at least print an error message and skip this entry or (my preferred solution) just exit with a non-zero exit code.